lib/xlibc-rt/check_stack.c

```diff
@@ -0,0 +1,25 @@
+#include <debug.h>
+#include "xboxkrnl/xboxkrnl.h"
+
+#ifdef DEBUG_CONSOLE
+    #define _print DbgPrint
+#else
+    #define _print debugPrint
+#endif
+
+void _xlibc_check_stack (unsigned int eax, unsigned int esp)
+{
+    // eax is the amount of requested memory
+    // esp is the stack before the call to __chkstk
+    PKTHREAD current_thread = KeGetCurrentThread();
+
+    if (esp - eax < (DWORD)current_thread->StackLimit || eax >= esp)
```

Comment:
nit: Might want to check `eax >= esp` first, before using them in the subtraction (which might give bogus results we don't care about)